### DISCUSSION TITLE
Learning Lab: Kong Custom Plugins

### DISCUSSION THREAD

**Discussion by pranitallam** [Kong/kong#9399] [Help]
In the learning lab environment for the creation of kong custom plugins(https://education.konghq.com/courses/course-v1:kong+KGLL-110+Perpetual/course)

 Once after adding the service, route and plugin, tried to hit the command http :8000/mock/anything Host:httpbin.org as suggested in course getting error as 404 not found , and not received the response as in the customised plugin.

I could see there is no mock  as I could see only  httpbin.org/anything URL, if I hit.

If I try to hit http :80/mock/anything Host:httpbin.org using container port 80, it is giving connection error.

Can you please suggest the functionality working here.

**Comment by samugi** [comment]
Ensure Kong is configured correctly, in particular, check the output of `http :8001/services/mock-service/routes` (assuming you used the name `mock-service` for your service) and confirm that your route contains both the host `httpbin.org` and the path `/mock`. 

Lastly, if the response contains the message `no Route matched with those values` the problem is that you are not hitting a valid route, while a more generic "404 - not found" is likely to be returned from the upstream (httpbin.org).
